set.mm

```diff
@@ -466289,6 +466442,482 @@ real number multiplication operation (this has to be defined in the main
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Permutation cycles
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c toCyc $.
+
+  $( Extend class notation with the permutation cycle builder. $)
+  ctocyc $a class toCyc $.
+
+  ${
+    $d d w u x $.
+    $( Define a convenience permutation cycle builder.  Given a list of
+       elements to be cycled, in the form of a word, this function produces the
+       corresponding permutation cycle.  See definition in [Lang] p. 30.
+       (Contributed by Thierry Arnoux, 19-Sep-2023.) $)
+    df-tocyc $a |- toCyc = ( d e. _V |-> ( w e.
+      { u e. Word d | u : dom u -1-1-> d } |-> ( ( _I |` ( d \ ran w ) )
```

Comment:
Why is it important/necessary that the word is injective, i.e. must consist of different symbols?